OpenEdXMobile/src/main/java/org/edx/mobile/player/IPlayerImpl.java

```diff
@@ -1,29 +1,43 @@
 package org.edx.mobile.player;
 
+import android.content.Context;
 import android.graphics.Point;
-import android.graphics.SurfaceTexture;
-import android.media.AudioManager;
-import android.media.MediaPlayer;
-import android.media.MediaPlayer.OnBufferingUpdateListener;
-import android.media.MediaPlayer.OnCompletionListener;
-import android.media.MediaPlayer.OnErrorListener;
-import android.media.MediaPlayer.OnInfoListener;
-import android.media.MediaPlayer.OnPreparedListener;
-import android.view.Surface;
-import android.view.TextureView;
+import android.net.Uri;
 import android.view.View.OnClickListener;
 
+import com.google.android.exoplayer2.ExoPlaybackException;
+import com.google.android.exoplayer2.ExoPlayerFactory;
+import com.google.android.exoplayer2.Player;
+import com.google.android.exoplayer2.SimpleExoPlayer;
+import com.google.android.exoplayer2.Timeline;
+import com.google.android.exoplayer2.analytics.AnalyticsListener;
+import com.google.android.exoplayer2.metadata.Metadata;
+import com.google.android.exoplayer2.metadata.MetadataOutput;
+import com.google.android.exoplayer2.source.ExtractorMediaSource;
+import com.google.android.exoplayer2.source.MediaSource;
+import com.google.android.exoplayer2.source.MediaSourceEventListener;
+import com.google.android.exoplayer2.source.hls.HlsMediaSource;
+import com.google.android.exoplayer2.ui.PlayerView;
+import com.google.android.exoplayer2.upstream.DataSource;
+import com.google.android.exoplayer2.upstream.DefaultDataSourceFactory;
+import com.google.android.exoplayer2.util.Util;
+
+import org.edx.mobile.R;
 import org.edx.mobile.logger.Logger;
+import org.edx.mobile.util.VideoUtil;
 import org.edx.mobile.view.OnSwipeListener;
 
-import java.io.File;
-import java.io.FileInputStream;
+import java.io.IOException;
 import java.util.Locale;
 
+import static com.google.android.exoplayer2.C.TRACK_TYPE_UNKNOWN;
+import static org.edx.mobile.util.AppConstants.VIDEO_FORMAT_M3U8;
+
 @SuppressWarnings("serial")
-public class Player extends MediaPlayer implements OnErrorListener,
-        OnPreparedListener, OnBufferingUpdateListener,
-        OnCompletionListener, OnInfoListener, IPlayer {
+public class IPlayerImpl implements Player.EventListener, MetadataOutput, AnalyticsListener, IPlayer {
```

Comment:
@miankhalid `Player` class name makes conflicts, because Exoplayer also have a class named `Player`, I think we need to update the `Player` to `PlayerListener` and IPlayerImpl to `VideoPlayer`, What do you think.

Reply:
Yep, PlayerListener and VideoPlayer sound good to me.